mod_wasm/modules/wasm/mod_wasm.c

```diff
@@ -170,17 +131,19 @@ static void *create_dir_config(apr_pool_t *p, char *dirspec)
      */
     cfg->local = 0;
     cfg->congenital = 0;
-    cfg->bEnableCGI = 0;
-    cfg->configEnvVarCount = 0;
-    cfg->configArgCount = 0;
+    cfg->bWasmEnableCGI = 0;
     cfg->cmode = CONFIG_MODE_DIRECTORY;
     /*
      * Finally, add our trace to the callback list.
      */
-    dname = (dname != NULL) ? dname : "";
-    cfg->loc = apr_pstrcat(p, "DIR(", dname, ")", NULL);
-    note = apr_psprintf(p, "create_dir_config(p == %pp, dirspec == %s)",
-                        (void*) p, dirspec);
+    context = (context != NULL) ? context : "";
+    cfg->loc = apr_pstrcat(p, "DIR(", context, ")", NULL);
+    note = apr_psprintf(p, "create_dir_config(p == %pp, context == %s)",
+                        (void*) p, context);
+
+    // creates a new Wasm config for the current context
+    wasm_config_create(cfg->loc);
```

Comment:
If this returned an error, we should handle it here.

Reply:
I agree